src/app/state/reducers/ethrdid.ts

```diff
@@ -0,0 +1,33 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { DIDDocument } from 'did-resolver'
+
+export interface EtherdidState {
+  didDocument: DIDDocument
```

Comment:
Now that we have only one entry in this reducer we can just make the state be that entry, no? I would call it the `didDocuement` reducer. It just contains the DID Document associated to the logged in user. I'm trying to keep this as simple and good-named as possible, extra complexity makes it less maintainable over time.

Reply:
Yes, this seems like it would work. However, with redux I am unable to replace the entire state with a new DIDDocument. The following always results in the initialState, not the new state with the DID doc:
```
reducers: {
    resolveDid (state: DIDDocument, { payload: data }: PayloadAction<DIDDocument>) {
      state = data
    }
  }
```


It seems there needs to be a root variable that gets assigned the DID document. I found this SO article:
https://stackoverflow.com/questions/60002846/how-can-you-replace-entire-state-in-redux-toolkit-reducer